alembic/versions/70f4a9879daf_add_miner_table.py

```diff
@@ -0,0 +1,193 @@
+"""add miner table
+
+Revision ID: 70f4a9879daf
+Revises: 9a64d1298abc
+Create Date: 2025-03-19 12:45:04.674078
+
+"""
+
+from typing import Sequence, Union
+
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision: str = "70f4a9879daf"
+down_revision: Union[str, None] = "9a64d1298abc"
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    op.create_table(
+        "miners",
+        sa.Column("id", sa.BigInteger(), nullable=False, autoincrement=True),
+        sa.Column("miner_uid", sa.Integer(), nullable=False),
+        sa.Column("coldkey", sa.String(), nullable=True),
+        sa.Column("hotkey", sa.String(), nullable=True),
+        sa.Column(
+            "created_at",
+            sa.DateTime(),
+            nullable=False,
+            server_default=sa.func.current_timestamp(),
+        ),
+        sa.Column(
+            "updated_at",
+            sa.DateTime(),
+            nullable=False,
+            server_default=sa.func.current_timestamp(),
+            server_onupdate=sa.func.current_timestamp(),
+        ),
+        sa.PrimaryKeyConstraint("id"),
+    )
+    op.create_index("miners_miner_uid", "miners", ["miner_uid"])
+    op.create_unique_constraint(
+        "uq_miners_miner_uid_coldkey_hotkey",
+        "miners",
+        ["miner_uid", "coldkey", "hotkey"],
```

Comment:
probably quite improbable, but it can happen that if I was deregistered from uid = 20 and then when I register again at later time I can acquire the same uid = 20 with the same coldkey and hotkey,
than there will be an error because of this restriction, right?

Reply:
good point, it's possible that with the same wallet I register uid 0, then 2 and then again 0, in which case it will not crash but we need to update the `updated_at` column so that we have the latest miners info.

So I will change to:
```
INSERT INTO miners (miner_uid, coldkey, hotkey, updated_at)
VALUES (0, '5DcqjtAz5zcLLbLA9BEKzv4oHkgpRrteBimA5J1wrhNKKjZf', '5HRPX6aZUdwiaFPKRWTgTbrtyKejFakWjKtXsPosdh3wRDv5', now())
    ON CONFLICT (miner_uid, coldkey, hotkey) DO
    UPDATE
    SET updated_at = now()
```